compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -198,6 +198,10 @@
     private static final ParserRuleContext[] CONST_DECL_RHS =
             { ParserRuleContext.STATEMENT_START_IDENTIFIER, ParserRuleContext.ASSIGN_OP };
 
+    private static final ParserRuleContext[] ARRAY_LENGTH = { ParserRuleContext.CLOSE_BRACKET,
+            ParserRuleContext.DECIMAL_INTEGER_LITERAL, ParserRuleContext.HEX_INTEGER_LITERAL,
+            ParserRuleContext.ASTERISK, ParserRuleContext.STATEMENT_START_IDENTIFIER};
```

Comment:
`STATEMENT_START_IDENTIFIER` should be replace with `VARIABLE_REF`.

Both indeed refers to identifiers, but the contexts are different..

Reply:
Done :-)